packages/runtime/test/integration/execute-transaction.ts

```diff
@@ -226,4 +251,264 @@ describe("Algorand Smart Contracts - Execute transaction", function () {
 			runtime.executeTx(tx);
 		}
 	});
+
+	it("Should opt-in and call app, through execute transaction", () => {
+		setupApp();
+		syncAccounts();
+
+		const appInfo = runtime.getAppInfoFromName(approvalProgramFilename, clearProgramFilename);
+		assert.isDefined(appInfo);
+		let tx: types.ExecParams[];
+		if (appInfo !== undefined) {
+			tx = [
+				{
+					type: types.TransactionType.OptInToApp,
+					sign: types.SignType.SecretKey,
+					fromAccount: alice.account,
+					appID: appInfo?.appID,
+					payFlags: { totalFee: 1000 },
+				},
+				{
+					type: types.TransactionType.CallApp,
+					sign: types.SignType.SecretKey,
+					fromAccount: alice.account,
+					appID: appInfo?.appID,
+					payFlags: { totalFee: 1000 },
+				},
+			];
+
+			runtime.executeTx(tx);
+		}
+
+		syncAccounts();
+		assert.equal(alice.balance(), initialBalance - 2000n);
+	});
```

Comment:
You should also check whether the account have opt-in the application or not

Reply:
ok. done

Reply:
Duplicated by mistake. Removed it.